vispy/gloo/tests/test_program.py

```diff
@@ -254,6 +254,13 @@ def test_varyings(self):
         # And anything else also fails
         self.assertRaises(KeyError, program.__getitem__, 'fooo')
 
+    def test_type_aliases(self):
+        program = Program("in bool A; out float B;", "foo")
+
+        # in aliased to attribute, out to varying
+        assert ('attribute', 'bool', 'A') in program.variables
```

Comment:
Is `'in'` also available in `.variables` or only the `attribute` version?

Reply:
only the attribute version; in the parsing function, `in` is only used for the regex, but immediately after everything is treated identically to `attribute`.